### DISCUSSION TITLE
Development TODO list

### DISCUSSION THREAD

**Discussion by Moohan** [Public-Health-Scotland/source-linkage-files#765] [General]
### SLF Issues

 * Preventable beddays  - #757 and #779 
 * Lots of blank datazone in A&E, due to PC8 postcode format #773 

### (re)Additions

 * HRI - #761 #778 
 * Homelessness flags - #771 
 * #777 #785

### September 23 Update

 * NSU cohort for 2022\23. - #768
 * Create the new 2023\24 files.
 * Final 'regular' update for 2021\22 files.

### (re) Additions

 * Death date fixes - #338 
 * keep_population flag - (#819). LIST analysts requesting this flag again, something to prioritise for the December update. 
 *  social care data
 * ICD10 codes for identifying drug related admissions 
 * review Tests workbook - split out episode/indiv file tests. 
 * review extract workbook tests 

### Code clean-up

 * Various code improvement suggestions - #686 #687 #725 
 * Remove redundant code (e.g. Individual file) #770 
 * Rename functions for consistency and clarity. Improve documentation. - #605 #580 #509 #508 

### Tests

 * Add missing tests (for data) - #671 #672
 * Improve existing (data) tests - #689 

### Improvements

 * `{keyring}` setup should be simpler - #545 
 * Split the tests for episode and individual out into their own workbooks.
 * Make the Social Care client info matching better - #107 # 106 #741 #742
 * Expand `person_id` to enable unique identification when records don't have a CHI - # 41 #714

## Longer term / On-hold

 * Bring in the `cup_marker` from the UCD to GP OoHs and Acute. #775 #776
 * Cross-year tests #11 
 * Review the cohorts - Create new / modify existing cohorts to take into account all available data.
 * Review DN / CMH now data collection has stopped (related #207)
 * NHS 24 added - #5 
 * SAS added - #4 
 * Add Ethnicity - #115

**Comment by Jennit07** [comment]
Hey @Moohan This looks great! Good to see everything in the one place to keep us on track! One thing that came to mind not on this list is a new NSU cohort and also new 23/24 file creation?

**Reply by Moohan** [reply]
Thanks, I've added those.
